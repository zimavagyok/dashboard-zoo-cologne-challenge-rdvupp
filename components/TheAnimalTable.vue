<script lang="ts" setup>
import { computed } from 'vue';
import type { PropType } from 'vue';
import type { Animal } from '../types';
import { calculateAgeInYears } from '../composables/helpers';

const props = defineProps({
  animals: {
    type: Array as PropType<Array<Animal>>,
    required: true,
  },
});

const animalsSortedByName = computed(() =>
  props.animals.slice().sort((animalA, animalB) => animalA.name > animalB.name ? 1 : -1)
);
</script>

<template>
  <table class="table-auto">
    <thead>
      <tr>
        <th>Index</th>
        <th>Species</th>
        <th>Name</th>
        <th>Gender</th>
        <th>Age (yrs)</th>
        <th>Weight (kg)</th>
      </tr>
    </thead>
    <tbody>
      <tr
        v-for="(
          { id, species, name, gender, birthdate, weight }, animalIndex
        ) in animalsSortedByName"
        :key="id"
        class="hover:bg-gray-200"
      >
        <td>{{ animalIndex + 1 }}</td>
        <td>{{ species }}</td>
        <td>{{ name }}</td>
        <td>{{ gender }}</td>
        <td>{{ calculateAgeInYears(new Date(birthdate)) }}</td>
        <td>{{ weight }}</td>
      </tr>
    </tbody>
  </table>
</template>

<style scoped>
table {
  @apply text-left;
}

td {
  @apply w-40;
}

tr {
  @apply border-b-2;
}
</style>
