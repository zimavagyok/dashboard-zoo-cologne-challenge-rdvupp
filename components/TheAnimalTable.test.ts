import { describe, expect, test } from 'vitest'

import { mount } from '@vue/test-utils'
import TheAnimalTable from './TheAnimalTable.vue'

describe('TheAnimalTable', () => {
  test('mounts and is still the same', () => {
    expect(TheAnimalTable).toBeTruthy()
    const wrapper = mount(TheAnimalTable, {
      props: {
        animals: [],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
