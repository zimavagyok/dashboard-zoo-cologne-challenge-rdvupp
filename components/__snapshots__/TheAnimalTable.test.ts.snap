// Vitest Snapshot v1

exports[`TheAnimalTable > mounts and is still the same 1`] = `
"<table class=\\"table-auto\\" data-v-5893c39c=\\"\\">
  <thead data-v-5893c39c=\\"\\">
    <tr data-v-5893c39c=\\"\\">
      <th data-v-5893c39c=\\"\\">Index</th>
      <th data-v-5893c39c=\\"\\">Species</th>
      <th data-v-5893c39c=\\"\\">Gender</th>
      <th data-v-5893c39c=\\"\\">Age (yrs)</th>
      <th data-v-5893c39c=\\"\\">Weight (kg)</th>
    </tr>
  </thead>
  <tbody data-v-5893c39c=\\"\\"></tbody>
</table>"
`;
